cri-o-testonly.spec

```diff
@@ -0,0 +1,138 @@
+# This file is only used for running automated rpm build tests for PRs
+# The canonical dist-git source can be found at:
+# https://src.fedoraproject.org/rpms/cri-o/blob/master/f/cri-o.spec
+
+%if 0%{?with_debug}
+%global _dwz_low_mem_die_limit 0
+%else
+%global debug_package   %{nil}
+%endif
+
+%global provider        github
+%global provider_tld    com
+%global project         kubernetes-incubator
+%global repo            cri-o
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path     %{provider_prefix}
+%global service_name crio
+
+Name:           cri-o-testonly
+Version:        1.0.0
+Release:        2.testonly%{?dist}
+Summary:        OCI-based implementation of Kubernetes Container Runtime Interface
+License:        ASL 2.0
+URL:            https://%{provider_prefix}
+Source0:        %{name}.tar.gz
+ExcludeArch:    ppc64
+# If go_compiler is not set to 1, there is no virtual provide. Use golang instead.
+BuildRequires:  %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires:  btrfs-progs-devel
+BuildRequires:  git
+BuildRequires:  glib2-devel
+BuildRequires:  glibc-static
+BuildRequires:  go-md2man
+BuildRequires:  gpgme-devel
+BuildRequires:  libassuan-devel
+BuildRequires:  libseccomp-static
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  device-mapper-devel
+BuildRequires:  make
```

Comment:
actually, I could add a `dnf builddep cri-o-testonly.spec`, but that needs to run as a root user. Let me know what kinda powers the bot/CI user would have, and I can add that command in as well as modify the `clean-rpm` target to whatever you like.

Reply:
Unf. it can be hard to say beyond "it depends".  Generally speaking, on the subject-VM itself, we're running as root or can easily gain root.  On the control-node, it's almost never root (though it could be).

My statement above was more in relation to the automation that could/would call 'make'.  If there's an easy way for that automation to extract all the ``BuildRequires`` for a given target (rhel or fedora or centos).  That would make it trivial for the automation to ensure the required packages are installed.  For example:

Automation running on subject VM:
1. ``make rhel-rpm-buildeps > required_rpms``
2. ``yum install -y required_rpms``
3. ``make rhel-rpm``

Human, maintaining ``.spec`` file:
1. Make changes to buildeps in one place
2. Push to a PR
3. Changes picked up and used by automation as above.

Note: None of this is actually required, or critical.  All in the nice-to-have category, and certainty doesn't need to be part of this PR.